source/common/stream_info/utility.cc

```diff
@@ -32,6 +32,41 @@ const std::string ResponseFlagUtils::DURATION_TIMEOUT = "DT";
 const std::string ResponseFlagUtils::UPSTREAM_PROTOCOL_ERROR = "UPE";
 const std::string ResponseFlagUtils::NO_CLUSTER_FOUND = "NC";
 
+const std::vector<std::pair<std::string, ResponseFlag>>
```

Comment:
non statically constructed non-PODs please.

Actually the above existing std::string declarations also violate this but I think you can replace them with absl::string_view which are safe to statically construct from string literals. See https://abseil.io/tips/140

I am not actually sure if it's kosher to make a static array of pairs. I suspect it's OK, but I'd at least follow the pattern in https://stackoverflow.com/questions/55099871/iinitializing-a-constexpr-stdarray-of-pairs and
 * use std::array rather than std::vector
 *  declare it constexpr, not const

Reply:
https://quuxplusone.github.io/blog/2018/06/26/cost-of-static-lifetime-constructors/ also makes interesting reading.

Reply:
neither string nor vector has constexpr constructor. Fortunately, pair is not a blocker.

array might resolve the vector, but string_view is not always a replacement since we need `const string&` to the string.
Let me give it another try to see if string_view with a limit amount `const string&`  to `string_view` works.

If it's too large I will use static strings variable within a function

Reply:
Yeah plus the current state is not IMO acceptable. We should not have static std::strings being constructed. in Envoy.

You may need to also change some existing APIs to take absl::string_view to make this all work.

Reply:
I'll just add also that absl::string_view was not available at the time some of the older Envoy code was written, so there are still some APIs that require `const std::string&`. But I think it's not too hard to get alterations of that form through review.